team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -91,8 +91,38 @@ private boolean playthrough(TreeNode current) {
 	}
 
 	// TODO
+
 	private TreeNode getMaxLeaf(TreeNode root) {
-		return null;
+		//Creating an array list of our leafNodes
+		ArrayList<TreeNode> leaves = getLeaves(root);
+
+		//Variables used to check against in the for loop
+		double check = leaves.get(0).getUCB();
+		int leafIndex = 0;
+
+		//Iterate through the array list of leaves finding the largest leaf.
+		for (int i = 1; i < leaves.size(); i++) {
+			if (leaves.get(i).getUCB() > check)
+				leafIndex = i;
+		}
```

Comment:
I don't think this code works at a glance. The `check` will never update. Also, maybe rename `check` to something more descriptive (i.e. `maxUCB`).